src/silx/gui/plot/test/testPlotWidget.py

```diff
@@ -2047,3 +2050,28 @@ class TestPlotMarkerLog_Gl(TestPlotMarkerLog):
 
 class TestSpecial_ExplicitMplBackend(TestSpecialBackend):
     backend = "mpl"
+
+
+@pytest.mark.parametrize("plotWidget", ("mpl", "gl"), indirect=True)
+@pytest.mark.parametrize(
+    "xerror,yerror",
+    [
+        (2, 2),  # Single value
+        ((1, 2, 3), (3, 2, 1)),  # Flat array
+        (([1], [2], [3]), ([3], [2], [1])),  # Nx1 array
+        ([(1, 2, 3), (3, 2, 1)], [(3, 2, 1), (1, 2, 3)]),  # 2xN array
+        (-1, -1),  # Negative values
+        ((-1, 0, 1), (1, 0, -1)),  # Flat array with negative values
```

Comment:
You might want to add inf, -inf and nan as well.

Reply:
Added some test cases with NaN and inf